CMakeLists.txt

```diff
@@ -27,6 +27,8 @@ if (MSVC)
         set (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} /MTd")
         set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /MTd")
     endif()
+ else()
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror=undef")
```

Comment:
Thanks for the PR. This line breaks the build, though. As you can see, gtest seems riddled with these. Also manipulating CXX_FLAGS is generally considered bad practice. Could you please add this to the `ZXING_CORE_LOCAL_DEFINES` in core/CMakeLists.txt and we see how far we get then?

Reply:
Ah, I hadn't noticed because unittests aren't built by default, and the name of the cmake option for building them is non-standard. It's `BUILD_UNIT_TESTS` while CMake standardizes the name `BUILD_TESTING` (see `Modules/CTest.cmake`).
Would it be OK for me to make a separate MR about changing that option name to use the standard one?
Then my build script would have enabled the tests automatically and I would have noticed this :)
But of course this would require updating existing scripts and CI on your side...

Reply:
Well, on the other hand, if I do that, I'll then hit the `-Werror=undef` in gtest itself, lol.... ;)

Reply:
The unit test option is not enabled by default because there are symbols exported to enable the unit testing which are not meant to be visible in a production build. (Might be a questionable approach but that is what it looks right now.)

And `BUILD_TESTING` is not a canonical name either, since we have two test sets: 'black box' tests that operate on example images and 'white box' tests, which are the gtest based 'unit tests' in this case.

Reply:
I completely agree that unittests shouldn't be enabled by default in production code.
Some projects enable them by default when selecting Debug mode. But no problem.
Exporting more symbols when unittests are enabled makes sense as well, e.g. Qt does that too.

Blackbox tests seem to be enabled by default, so maybe `BUILD_TESTING` could be used for the unittests...